jgrapht-core/src/main/java/org/jgrapht/alg/interfaces/PartitionAlgorithm.java

```diff
@@ -0,0 +1,95 @@
+/*
+ * (C) Copyright 2018-2018, by Alexandru Valeanu and Contributors.
+ *
+ * JGraphT : a free Java graph-theory library
+ *
+ * This program and the accompanying materials are dual-licensed under
+ * either
+ *
+ * (a) the terms of the GNU Lesser General Public License version 2.1
+ * as published by the Free Software Foundation, or (at your option) any
+ * later version.
+ *
+ * or (per the licensee's choosing)
+ *
+ * (b) the terms of the Eclipse Public License v1.0 as published by
+ * the Eclipse Foundation.
+ */
+package org.jgrapht.alg.interfaces;
+
+import java.util.Collections;
+import java.util.Set;
+
+/**
+ * Algorithm to compute a vertex partition of a graph.
+ *
+ * @param <V> vertex the graph vertex type
+ *
+ * @author Alexandru Valeanu
+ * @since June 2018
+ */
+public interface PartitionAlgorithm<V> {
```

Comment:
We should make this interface more generic, to support k-way partitioning. I guess the partitions should be represented as a map from vertices to integers. Additionally we could also have a special  interface for 2-way partitioning.

Reply:
What about a list of sets? A map from vertices to integers would have to somehow restrict the values to [0..k-1] (or hope that that user doesn't break it somehow).

Reply:
The previous attempt on bipartite partitioning used the VertexColoringAlgorithm interface, which already provides both representations (list of sets; map from vertex to integer).  In general, a partittioning is not the same as a coloring (since coloring implies a constraint on the edges, whereas partitioning is arbitrary), but we might want to think the relationship through.

Reply:
@jsichi @d-michail I have rewritten the interface based on the `VertexColoringAlgorithm` interface.